woocommerce/woocommerce-gutenberg-products-block
# Graduate `__experimental_woocommerce_blocks_checkout_update_order_from_request` action to stable

This PR will deprecate the `__experimental_woocommerce_blocks_checkout_update_order_from_request` action and replace it with `woocommerce_blocks_checkout_update_order_from_request`. 

It also updates the docs relating to this hook. All references to the now deprecated hook have been _replaced_ with the new hook, besides the one in the experimental interfaces document, which has changed to note that the prefixed hook is now deprecated.


Fixes #4997 

### Manual Testing

How to test the changes in this Pull Request:

1. Add the following code somewhere that it'll be executed, for example your theme's `functions.php` file:
```php
add_action( '__experimental_woocommerce_blocks_checkout_update_order_from_request', function( $order, $request ) { return; }, 10, 2 );

add_action( 'woocommerce_blocks_checkout_update_order_from_request', function( $order, $request ) {
	// For testing purposes, set the customer note to the user-agent from the request.
	$order->set_customer_note( $request->get_header('User-agent') );
	$order->save();
}, 10, 2 );
```
2. Add the following directives to your `wp-config.php` file:
```php
define( 'WP_DEBUG', true );
define( 'WP_DEBUG_LOG', true );
```
3. Add a product to your cart and check out using the Checkout block. Ensure it goes through correctly.
4. Check your `debug.log` file and ensure there's a deprecation message logged for `__experimental_woocommerce_blocks_checkout_update_order_from_request`.
5. Ensure the order contains a customer note that is your browser's user agent.



### Changelog

> Deprecate the __experimental_woocommerce_blocks_checkout_update_order_from_request action in favour of woocommerce_blocks_checkout_update_order_from_request.
